content/en/blog/_posts/XXXX-XX-XX-fine-grained-supplementalgroups-control-beta/index.md

```diff
@@ -0,0 +1,195 @@
+
+Let's see an example, below Pod specifies `runAsUser=1000`, `runAsGroup=3000` and `supplementalGroups=4000` in the Pod's security context.
+
+{{% code_sample file="implicit-groups.yaml" %}}
+
+What is the result of `id` command in the `ctr` container?
+
+```console
+# Create the Pod:
+$ kubectl apply -f https://k8s.io/blog/XXXX/XX/XX/Fine-grained-SupplementalGroups-control/implicit-groups.yaml
+
+# Verify that the Pod's Container is running:
+$ kubectl get pod implicit-groups
+
+# Check the id command
+$ kubectl exec implicit-groups -- id
+```
+
+Then, output should be similar to this:
+
+```none
+uid=1000 gid=3000 groups=3000,4000,50000
+```
+
+Where does group ID `50000` in supplementary groups (`groups` field) come from, even though `50000` is not defined in the Pod's manifest at all? The answer is `/etc/group` file in the container image.
+
+Checking the contents of `/etc/group` in the container image should show below:
+
+```console
+$ kubectl exec implicit-groups -- cat /etc/group
+...
+user-defined-in-image:x:1000:
+group-defined-in-image:x:50000:user-defined-in-image
+```
+
+Aha! The container's primary user `1000` belongs to the group `50000` in the last entry.
+
+Thus, the group membership defined in `/etc/group` in the container image for the container's primary user is _implicitly_ merged to the information from the Pod. Please note that this was a design decision the current CRI implementations inherited from Docker, and the community never really reconsidered it until now.
+
+### What's wrong with it?
+
+The _implicitly_ merged group information from `/etc/group` in the container image may cause some concerns particularly in accessing volumes (see [kubernetes/kubernetes#112879](https://issue.k8s.io/112879) for details) because file permission is controlled by uid/gid in Linux. Even worse, the implicit gids from `/etc/group` can not be detected/validated by any policy engines because there is no clue for the implicit group information in the manifest. This can also be a concern for Kubernetes security.
+
+## Fine-grained SupplementalGroups control in a Pod: `SupplementaryGroupsPolicy`
+
+To tackle the above problem, Pod's `.spec.securityContext` now have `supplementalGroupsPolicy` field.
+
+This field provides a way to control how to calculate supplementary groups for the container processes in a Pod. The available policy is below:
+
+* _Merge_: The group membership defined in `/etc/group` for the container's primary user will be merged. If not specified, this policy will be applied (i.e. as-is behavior for backward compatibility).
+
+* _Strict_: it only attaches specified group IDs in `fsGroup`, `supplementalGroups`, or `runAsGroup` fields as the supplementary groups of the container processes. This means no group membership defined in `/etc/group` for the container's primary user will be merged.
```

Comment:
should we say that this is a recommended option and we hope all Pod will use it in a few years?

Reply:
Yes in terms of security, personally. Probably, bunch of cluster admins does not aware the "implicit" gids from the container images.

Although it is not strong recommendation, docs for this feature includes a pointer to the risk of _implicitly merged_ gids:

https://github.com/kubernetes/website/blob/7a9e33eb515a1afc2005fdef2127a701ef3548bb/content/en/docs/tasks/configure-pod-container/security-context.md?plain=1#L227-L231

Reply:
Making sure the sentence follows the above "Merge" policy, and use "ignored" to
make it clear that this will remove some default user setup

```suggestion
* _Strict_: Only the group IDs specified in `fsGroup`, `supplementalGroups`, or `runAsGroup` are attached as supplementary groups to the container processes. Group memberships defined in `/etc/group` for the container's primary user are ignored.
```